columnphysics/icepack_shortwave.F90

```diff
@@ -4815,22 +4810,38 @@ subroutine compute_dEdd_5bd(                          &
                   ks = ext_cff_mss_ice_drc(ns,nmbrad_snicar)
                   ws = ss_alb_ice_drc     (ns,nmbrad_snicar)
                   gs = asm_prm_ice_drc    (ns,nmbrad_snicar)
-               elseif (ceiling(rsnw(ksnow)) - rsnw(ksnow) < 1.0e-3_dbl_kind) then
-                  ! radius = 30 --> nr = 1 in SNICAR table      ! NOTE
-                  nr = ceiling(rsnw(ksnow)) - 30 + 1            ! hardwired min radius = 30
-                  ks = ext_cff_mss_ice_drc(ns,nr)
-                  ws = ss_alb_ice_drc     (ns,nr)
-                  gs = asm_prm_ice_drc    (ns,nr)
-               else ! linear interpolation in rsnw
-                  nr = ceiling(rsnw(ksnow)) - 30 + 1            ! hardwired min radius = 30
-                  delr = (rsnw(ksnow) - floor(rsnw(ksnow))) &   ! hardwired delta radius = 1 in table
-                       / (ceiling(rsnw(ksnow)) - floor(rsnw(ksnow))) ! denom always = 1
-                  ks = ext_cff_mss_ice_drc(ns,nr-1)*(c1-delr) &
-                     + ext_cff_mss_ice_drc(ns,nr  )*    delr
-                  ws = ss_alb_ice_drc     (ns,nr-1)*(c1-delr) &
-                     + ss_alb_ice_drc     (ns,nr  )*    delr
-                  gs = asm_prm_ice_drc    (ns,nr-1)*(c1-delr) &
-                     + asm_prm_ice_drc    (ns,nr  )*    delr
+               else
+                  if (trim(snw_ssp_table) == 'snicar') then
+                     ! NOTE:  Assumes delta rsnw is 1 and rsnw are basically integers
+                     nr = ceiling(rsnw(ksnow)) - nint(rsnw_snicar_min) + 1
+                     if (ceiling(rsnw(ksnow)) - rsnw(ksnow) < 1.0e-3_dbl_kind) then
+                        ks = ext_cff_mss_ice_drc(ns,nr)
+                        ws = ss_alb_ice_drc     (ns,nr)
+                        gs = asm_prm_ice_drc    (ns,nr)
+                     else
+                        ! linear interpolation
+                        delr = (rsnw(ksnow) - floor(rsnw(ksnow))) &   ! hardwired delta radius = 1 in table
+                             / (ceiling(rsnw(ksnow)) - floor(rsnw(ksnow))) ! denom always = 1
+                        ks = ext_cff_mss_ice_drc(ns,nr-1)*(c1-delr) &
+                           + ext_cff_mss_ice_drc(ns,nr  )*    delr
+                        ws = ss_alb_ice_drc     (ns,nr-1)*(c1-delr) &
+                           + ss_alb_ice_drc     (ns,nr  )*    delr
+                        gs = asm_prm_ice_drc    (ns,nr-1)*(c1-delr) &
+                           + asm_prm_ice_drc    (ns,nr  )*    delr
+                     endif
+                  else
+                     ! linear interpolation
```

Comment:
My goal here was to leave the "snicar" snw table results bit-for-bit and add support for the "test" table.  That meant leaving all the logic and math as implemented.  Changing the logic or the math could/would change answers.  I agree that we could (should!) clean this up, but I propose it be done as a separate PR which would allow us to more carefully test the snicar results.  This PR, to some degree, was just to add support for the "test" table.